app/services/keybase_service.rb

```diff
@@ -0,0 +1,27 @@
+class KeyBaseService
+
+  include AgentLogic
+
+  def initialize
+    @keybase_url = "https://keybase.io/"
+  end
+
+  def get_validator_avatar(validator)
+    image_css = '.picture > .img-circle'
+    keybase_url = @keybase_url + validator.keybase_id
+    keybase = get_page(url: keybase_url)
+
+    if keybase[:http_code] == '200'
+
+      if Nokogiri::HTML(keybase[:html]).css(image_css).present?
+        avatar_url = Nokogiri::HTML(keybase[:html]).css(image_css)
+                            .attribute('src').value
+        if avatar_url.to_s.include?('keybase_processed_uploads')
+          puts "update #{validator.name} avatar to #{avatar_url}"
+          validator.update(avatar_url: avatar_url)
+        end
+      end
+    end
```

Comment:
If the call to Keybase fails, let's update the validator record with the default avatar URL at 'https://keybase.io/images/no-photo/placeholder-avatar-180-x-180@2x.png'

Reply:
I'm afraid that if I do that, we'll never catch the future updates of this url. maybe I should clear it instead?

Reply:
Hmmm... We're using a null value in the avatar_url column to determine new v. existing validators. The daily update script will pick up changes if the validator later provides a photo to Keybase.

It sounds like you are concerned that Keybase might change the location of their default image and then we will show a bunch of broken image links. We can put a placeholder image on our site and link to the local URL instead when we need a placeholder image. Sound good?

Reply:
Yes, but should I also check for the newer version of the avatar if validator has default url in our db ? (this one should not be broken) And if so, shouldn't we be worried about requesting keybase too much ?

Reply:
Yes, the daily update script should check Keybase for new URLs for all validators. Replace our default if we see a URL on Keybase. Add a 1-second sleep between requests on the daily script so we don't hammer their servers.